### ISSUE TITLE
Cliché detector: passive-voice offsets point at the wrong words, "was not" is flagged as passive, and inflected clichés are missed

### ISSUE THREAD

**Issue by MOHITKOURAV01** [Muskankr/AI-Resume-Analyzer#938]
`analyzer/cliche_detector.py` returns character offsets that don't point at the text they describe, flags ordinary sentences as passive, and misses the inflected forms of the clichés it already knows about.

## 1. Passive-voice offsets always point at the first occurrence

`detect_passive_voice` walks the words by index but then recovers the position with `find`, which returns the *first* match in the string rather than the one it is currently looking at:

```python
"start": text.lower().find(f"{word_lower} {next_word}"),
```

```python
>>> t = "Was promoted twice. The report was promoted again later."
>>> [(d["phrase"], d["start"], d["end"]) for d in detect_passive_voice(t)]
[('Was promoted', 0, 12), ('was promoted', 0, 12)]
```

Two distinct occurrences, one span. The second is 31 characters away from where it is reported. `ClicheDetector.tsx` uses these offsets to highlight the text, so the wrong words get highlighted — and any repeated phrase highlights the same place twice.

## 2. `endswith("t")` is not a past-participle test

```python
if next_word.endswith(("ed", "en", "t")):
```

`"t"` matches an enormous number of ordinary words:

```python
>>> [d["phrase"] for d in detect_passive_voice("This was not part of the team and is best in class")]
['was not', 'is best']
```

"was not", "is best", "was at", "was part", "is it", "was about" — none are passive voice. Every one of them costs the user 5 points of impact score and shows up as a highlighted "issue" in the UI.

## 3. `PASSIVE_INDICATORS` is dead code

The module defines a `PASSIVE_INDICATORS` list of regexes at the top and never references it. `detect_passive_voice` hardcodes its own copy of the same six words inline. Two sources of truth, one of which does nothing.

## 4. Inflected clichés are missed

The dictionary has `\bhandled\b`, so "handled" is caught but "handling" and "handles" are not. Same for the rest of the verb entries.

```python
>>> analyze_and_suggest("Tasked with handling the database migration.")["modernized_text"]
'executed handling the database migration.'
```

`test_analyze_and_suggest_modernization` fails here — it asserts "handling" is gone.

## 5. The modernizer can produce passive voice

`\bresponsible for\b → spearheaded` is applied without looking at what precedes it:

> "I was responsible for the rollout" → "I was spearheaded the rollout"

A tool whose stated job is removing passive voice should not introduce it.

## 6. Replacements lose capitalisation on all-caps input

`original[0].isupper()` only checks the first character, so `"RESPONSIBLE FOR"` becomes `"Spearheaded"`. Minor, but it silently changes a heading's style.

## 7. The score punishes short bullets

```python
score = max(0, 100 - int((issues_count / max(total_words, 1)) * 500))
```

One detection in an 8-word bullet scores 38/100. Because of #2 that detection is frequently a false positive on "was not".

## Note on the existing tests

`test_detect_cliches_case_insensitive` asserts the detected phrase for `"I Was Responsible For the project."` is `"Was Responsible For"`, but the pattern is `\bresponsible for\b` and does not include "was" — `test_detect_cliches_finds_exact_match` asserts `"responsible for"` for the same construction one test earlier. The two assertions contradict each other; the second one is the wrong one.

## Expected

Offsets that point at the occurrence being reported, a participle test that doesn't fire on "was not", one source of truth for the indicator list, inflection coverage, and a modernizer that doesn't write passive voice.
